src/pages/CommentsPage.jsx

```diff
@@ -1,34 +1,73 @@
 import { useState, useEffect } from "react";
-import { useNavigate, useParams } from "react-router-dom";
-import { getArticleComments } from "../utils/api";
+import { Link, useParams } from "react-router-dom";
+import { getArticleComments, postComment } from "../utils/api";
 import Loading from "../components/Loading";
 import CommentCard from "../components/CommentCard";
-import CommentForm from "../components/CommentForm";
 
 const CommentsPage = () => {
   const { article_id } = useParams();
   const [comments, setComments] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
-  const navigate = useNavigate();
+  const [input, setInput] = useState("");
+  const [err, setErr] = useState(null);
+
+  const updateInput = (e) => {
+    setInput(e.target.value);
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setIsLoading(true);
+    const postBody = { username: "jessjelly", comment: input };
+    postComment(article_id, postBody)
+      .then((res) => {
+        const { postedComment } = res.data;
+        setComments((currComments) => {
+          return [...currComments, postedComment];
+        });
+        setIsLoading(false);
+      })
+      .catch((err) => {
+        setErr("Something went wrong, please try again.");
+      });
+
+    setInput("");
+  };
 
   useEffect(() => {
     getArticleComments(article_id).then((res) => {
       const { comments } = res.data;
       setComments(comments);
       setIsLoading(false);
     });
-  }, []);
-
-  const handleBackToArticle = () => {
-    navigate(`/articles/${article_id}`);
-  };
+  }, [handleSubmit]);
```

Comment:
this is making 100s of requests every minute because the useEffect isn't supposed to take a function here - it doesn't even need to, because you're handling showing the new comment in handleSubmit.